flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,190 @@ import Sentry
-          SentrySDK.start { options in
-              options.dsn = dsn as? String
-              options.debug = resource["SentryDebug"] as? Bool ?? false
-              options.attachStacktrace = true
-          }
-        }
-      }
-    }
-  }
+    var options: Options?
+    
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
+        let instance = SwiftSentryFlutterPlugin()
+        registrar.addMethodCallDelegate(instance, channel: channel)
+    }
+
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method as String{
+            case "initNativeSdk" :
+            initNativeSdk(call, result: result)
+                break
+
+        case "captureEnvelope" :
+            captureEnvelope(call, result: result)
+            break
+        default:
+            result(FlutterMethodNotImplemented)
+        }
+
+
+    }
+
+    private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult){
+        guard let arguments = call.arguments as? [String: Any] else {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
+
+        if (arguments.isEmpty) {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
+
+        SentrySDK.start { options in
+            if(arguments["dsn"] != nil ){
+                options.dsn = arguments["dsn"] as? String
+            } else {
+                result(FlutterError(code: "4", message: "A valid Dsn must be provided", details: nil) )
+            }
+            if let isDebug = arguments["debug"] as? Bool {
+                options.debug = isDebug
+            }
+
+            if let environment = arguments["environment"] as? String{
+                options.environment = environment
+            }
+            if let releaseName = arguments["release"] as? String{
+                options.releaseName = releaseName
+            }
+
+            if let enableAutoSessionTracking = arguments["enableAutoSessionTracking"] as? Bool {
+                options.enableAutoSessionTracking = enableAutoSessionTracking
+            }
+
+            if let attachStacktrace = arguments["sessionTrackingIntervalMillis"] as? Bool {
+                options.attachStacktrace = attachStacktrace
+            }
+
+            if let sessionTrackingIntervalMillis = arguments["sessionTrackingIntervalMillis"] as? UInt{
+                options.sessionTrackingIntervalMillis = sessionTrackingIntervalMillis
+            }
+
+            if let dist = arguments["dist"] as? String{
+                options.dist = dist
+            }
+
+            if let integrations = arguments["integrations"] as? [String]{
+                options.integrations = integrations
```

Comment:
maybe we should append?

Reply:
✅

Reply:
those integrations are not the same as the `sdk.integrations`

the gotten `integrations` should be appended on the beforeSend callback,
this also lives in the event.sdk.integrations